nomad/structs/structs.go

```diff
@@ -5410,6 +5410,22 @@ func (t *Task) Validate(ephemeralDisk *EphemeralDisk, jobType string) error {
 		}
 	}
 
+	// Validation for Kind field which is used for Consul Connect integration
+	// TODO better wording for all error messages
+	taskKind := t.Kind
+	if taskKind.IsConnect() {
+		// This task is a Connect proxy so it should not have service stanzas
+		if len(t.Services) > 0 {
+			mErr.Errors = append(mErr.Errors, fmt.Errorf("Connect proxy task should not have service stanza in it"))
```

Comment:
Matching phrasing of next error

```suggestion
			mErr.Errors = append(mErr.Errors, fmt.Errorf("Connect proxy task must not have a service stanza"))
```